### PR TITLE
Update toolchain to 2022-07-05

### LINKED ISSUE #1336: Update rust toolchain version for Kani 0.6
We update the rust toolchain version once every release. For 0.6, we will try to update to version nightly-2022-07-05.

### PR DESCRIPTION
**Pull request by celinval**
### Description of changes: 

Update toolchain to 2022-07-05

The updates required were related to the following changes:
  -  https://github.com/rust-lang/rust/pull/97423
  -  https://github.com/rust-lang/rust/pull/98165
  -  https://github.com/rust-lang/rust/pull/97935
  -  https://github.com/rust-lang/rust/pull/98098
  -  https://github.com/rust-lang/rust/pull/96591

### Resolved issues:

Resolves #1336 

### Testing:

* How is this change tested? Existing tests

* Is this a refactor change? No

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs [hunk 1]
```diff
@@ -123,12 +122,22 @@ impl<'tcx> GotocCtx<'tcx> {
...
+                        let typ = self.codegen_ty(lit_ty);
+                        return self.codegen_unimplemented(
+                            "Constant slice value with 2+ bytes",
+                            typ,
+                            loc,
+                            "https://github.com/model-checking/kani/issues/1339",
+                        );
                     }
                 }
                 _ => {}
```
Comment (danielsn) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:137]: What does this do?

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs [hunk 2]
```diff
@@ -123,12 +122,22 @@ impl<'tcx> GotocCtx<'tcx> {
                             len_expr,
                             &self.symbol_table,
                         );
+                    } else {
+                        // TODO: Handle cases with other types such as tuples and larger integers.
+                        let loc = span.map_or(Location::none(), |s| self.codegen_span(s));
```
Comment (danielsn) [inline kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:127]: I think there is a `codegen_span_option` function that does this.

File: tests/kani/Intrinsics/Atomic/Unstable/AtomicMin/main.rs [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
-// Check that `atomic_min` and other variants (unstable version) return the
+// Check that `atomic_min_seqcst` and other variants (unstable version) return the
 // expected result.
 
 #![feature(core_intrinsics)]
 use std::intrinsics::{
-    atomic_min, atomic_min_acq, atomic_min_acqrel, atomic_min_rel, atomic_min_relaxed,
+    atomic_min_acqrel, atomic_min_acquire, atomic_min_relaxed, atomic_min_release,
```
Comment (danielsn) [inline tests/kani/Intrinsics/Atomic/Unstable/AtomicMin/main.rs:9]: Looks like the new namespace may include more options for atomic intrinsics: something for @adpaco-aws to look at?

File: kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs (hunk 1 above)
Comment (celinval) [reply kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs:137]: I believe this is a no-op. In this case, the `unimplemented!()` macro will be invoked at the end.

**Review by tedinski — APPROVED** [review]
DSN had a minor suggestion I think, then lgtm.

File: tests/kani/Intrinsics/Atomic/Unstable/AtomicMin/main.rs (hunk 3 above)
Comment (celinval) [reply tests/kani/Intrinsics/Atomic/Unstable/AtomicMin/main.rs:9]: I created https://github.com/model-checking/kani/issues/1344 for that.

**Pull request merged**